### PR TITLE
Inline documentation for public interfaces

### PR DESCRIPTION
**Pull request by DerekLimble**
Also...
- improves some import/exports
- removes the `type` method from events
- Other small improvements and cleanups

### CODE REVIEW

File: projects/limble-tree/src/lib/core/tree-branch/tree-branch.ts [hunk 1]
```diff
@@ -66,10 +83,18 @@ export class TreeBranch<UserlandComponent>
       }
    }
 
+   /** Returns all child branches as an array of TreeBranch instances */
```
Comment (dreamlaneLimble) [inline projects/limble-tree/src/lib/core/tree-branch/tree-branch.ts:86]: nit: here and elsewhere you can use the TSDoc `@returns`

File: projects/limble-tree/src/lib/core/tree-branch/tree-branch.ts [hunk 2]
```diff
@@ -83,23 +108,43 @@ export class TreeBranch<UserlandComponent>
...
+   /**
+    * Emits the specified TreeEvent.
+    *
+    * Caution: It is not recommended to manually emit TreeEvents that are already
+    * provided by the library. For example, it is not recommended to emit a
+    * `GraftEvent`, `DestructionEvent`, etc. These events may be used by the tree,
+    * and emitting them manually may cause unexpected behavior. Instead, we
+    * recommend implementing the TreeEvent interface with your own custom events
+    * and dispatching those.
+    */
```
Comment (dreamlaneLimble) [inline projects/limble-tree/src/lib/core/tree-branch/tree-branch.ts:125]: nit: This public function has a parameter, and you can use `@param` here to specify the purpose of the param.

Something like `@param event - The {@link TreeEvent} that will be emitted.`

**Review by dreamlaneLimble — COMMENTED** [review]
Looks great! Approved.

I added some nit comments that you can ignore or go back through and apply. We aren't specifically using TSDoc, so that's why they are nits, but sometimes it is nice to use the TSDoc style markup.

**Review by dreamlaneLimble — APPROVED** [review]

**Comment by DerekLimble** [comment]
Good feedback. I had considered adding tsdoc tags to be an "extra mile" activity when I wrote the doc blocks, but I am quite happy to go that extra mile now! I have updated all of the new doc blocks to use tagged sections.

**Review by dreamlaneLimble — APPROVED** [review]

**Pull request merged**